components/core/src/streaming_archive/writer/Archive.hpp

```diff
@@ -148,6 +135,12 @@ namespace streaming_archive { namespace writer {
 
         size_t get_data_size_of_dictionaries () const { return m_logtype_dict.get_data_size() + m_var_dict.get_data_size(); }
 
+        /**
+         * Sets the split status of the current encoded file
+         * @param is_split
+         */
+        void set_file_is_split (bool is_split) { m_file->set_is_split(is_split); };
```

Comment:
* Let's group all the wrappers together.
* Also, let's check `if (nullptr == m_file) { throw ... }` in all the wrappers.
* Since the wrappers will no longer be one-liners, you can move all of them to the `.cpp`